core/modes/modes.go

```diff
@@ -78,12 +78,22 @@ func ReconstructRequest(pair models.RequestResponsePair) (*http.Request, error)
 }
 
 // ReconstructResponse changes original response with details provided in Constructor Payload.Response
-func ReconstructResponse(request *http.Request, pair models.RequestResponsePair) *http.Response {
+func ReconstructResponse(request *http.Request, pair models.RequestResponsePair) (*http.Response, error) {
```

Comment:
This function is invoked on every request after the response is fetched from a cache and middleware has been applied. Doing file reading here would be a performance killer.

Ideally the response body should be loaded from the file when it gets imported, it also fails fast if the file is not readable. 

I would suggest putting this logic in `payload.go` -> `NewResponseDetailsFromResponse`